Library/Homebrew/utils.rb

```diff
@@ -291,6 +291,16 @@ def paths
   end.uniq.compact
 end
 
+# return the shell profile file based on users' preference shell
+def shell_profile
+  case ENV["SHELL"]
+  when %r{/(ba)?sh} then "~/.bash_profile"
+  when %r{/zsh} then "~/.zshrc"
+  when %r{/ksh} then "~/.kshrc"
+  else "~/.profile"
```

Comment:
Yeh, I know, I also use `zsh`. This is in the case that `SHELL` is unset or not matching something we recognise. I think in that situation it's worth assuming the default, Bash case rather than something else that doesn't work in all shells.